evm/src/executor/inspector/cheatcodes/ext.rs

```diff
@@ -76,6 +78,80 @@ fn get_code(path: &str) -> Result<Bytes, Bytes> {
     }
 }
 
+fn set_env(key: &str, val: &str) -> Result<Bytes, Bytes> {
+    // `std::env::set_var` may panic in the following situations
+    // ref: https://doc.rust-lang.org/std/env/fn.set_var.html
+    if key.is_empty() {
+        Err("Environment variable key can't be empty".to_string().encode().into())
+    } else if key.contains('=') {
+        Err("Environment variable key can't contain equal sign `=`".to_string().encode().into())
+    } else if key.contains('\0') {
+        Err("Environment variable key can't contain NUL character `\\0`"
+            .to_string()
+            .encode()
+            .into())
+    } else if val.contains('\0') {
+        Err("Environment variable value can't contain NUL character `\\0`"
+            .to_string()
+            .encode()
+            .into())
+    } else {
+        env::set_var(key, val);
+        Ok(Bytes::new())
```

Comment:
I don't think we have a custom `bail!` macro yet that does 
`return Err($msg.to_string().encode().into())`

would be a nice to have imo

Reply:
would probably do this in a separate PR, to deal with this boilerplate code scattered around the `cheatcodes/` codebase